clang/include/clang/Sema/Sema.h

```diff
@@ -11682,6 +11683,7 @@ class Sema final : public SemaBase {
                         SmallVectorImpl<TemplateArgument> &SugaredConverted,
                         SmallVectorImpl<TemplateArgument> &CanonicalConverted,
                         CheckTemplateArgumentKind CTAK, bool PartialOrdering,
+                        bool PartialOrderingTTP,
```

Comment:
Find myself wishing we could combine this with at least `PartialOrdering` in an enum of some sort.

Reply:
In this case, they are orthogonal and all four combinations are valid. Maybe renaming s/PartialOrderingTTP/MatchingTTP would make that more clear, but there are tons of pre-existing users/

Reply:
Thats always the balance unfortunately.  I hate having 2 bools in a row like this, particularly when they are both SO similar in name/meaning.  I would still love an enum with 4 values, but I can wait to see if others are as concerned as I am, and disagree and commit if necessary.

Reply:
We can turn the two bools into enums, and we can group parameters which often go together into an 'Info' like thing. Grouping ortogonal bools into one enum like that will create friction in that we will often want to combine and split them apart anyway.

Reply:
Ping, how may we proceed?

I don't think a good solution to this would come in this patch, as we have the same problem in multiple places, and this would add a lot of noise here.

We are due a refactor of these partial ordering functions, and I'd prefer to tackle that on a followup.

Reply:
Done on top of the patch stack: https://github.com/llvm/llvm-project/pull/124668